src/lib.rs

```diff
@@ -1,35 +1,109 @@
-//! This is the documentation for the rust-abci crate.
-#![feature(ptr_internals)]
+//! # Tendermint ABCI library for Rust
+//!
+//! This library provides an application Trait and TCP server for implementing Tendemint ABCI
+//! application in Rust.  The Application Trait provides default implementations for each callback
+//! to simplify development.
+//!
+//! ## Example
+//!
+//! Here's a simple example that communicates with Tendermint. Defaults callbacks are handled by
+//! the Trait.  The app doesn't do any actual processing on a transaction.
+//!
+//! ```
+//! struct EmptyApp;
+//!
+//! impl abci::Application for EmptyApp {}
+//!
+//! abci::run_local(EmptyApp);
```

Comment:
Put the `abci::run_local` call inside a `main` function for this example, because otherwise doc examples will fail to build.